### PR TITLE
Initial (draft) PR for edit mission as organizer

### PR DESCRIPTION
**Pull request by laredotornado**
Here is figma card -- https://www.figma.com/file/GLg0zPNnwMEGIb9TytCrxQ/Resilience-App?node-id=29813%3A13788

and AirTable card -- https://airtable.com/tblRjd2McKgRW9MKj/viwlRRNI0PZIxrGkc/recUd4ZiLI1THmka5

#431

### CODE REVIEW

File: firepit-log.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Welcome to firepit v1.1.0!
```
Comment (qdozaq) [inline firepit-log.txt:1]: Can add this file to `.gitignore`

File: src/app/dashboard/Missions/Missions.jsx [hunk 2]
```diff
@@ -117,7 +117,7 @@ const DashboardMissions = ({ inDone, inPlanning, inProgress, inProposed, volunte
   const [selectedMission, setSelectedMission] = useState(null);
 
   const all = { inProposed, inPlanning, inProgress, inDone };
-  const filtered = all[viewFromUrl] || inProposed;
+  const filtered = all["inProposed"] || inProposed;
```
Comment (qdozaq) [inline src/app/dashboard/Missions/Missions.jsx:120]: Just testing here?

File: src/app/dashboard/Missions/MissionEditView.jsx [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+    const { location } = data;
+    changeFormValue("location", location);
+  }
+
+  function handleChangePickUpLocationLabel(e) {
+    e.preventDefault();
+    changeFormValue("pickUpLocationLabel", e.target.value);
+  }
+
+  function handleChangeRecipientPhoneNumber(e) {
```
Comment (qdozaq) [inline src/app/dashboard/Missions/MissionEditView.jsx:220]: fyi you only need to make these functions for non standard inputs.

So normally you can just do
```js
<Input
name ='phone'
onChange={handleChange}
>
{values.phone}
</Input>
```

File: src/app/dashboard/Missions/MissionEditView.jsx [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+          </Grid>
+        </Grid>
+      ))}
+    </Box>
+  );
+};
+
+const MissionDetailsRow = ({ classes, mission }) => {
+  let type = mission?.type;
+  let details = mission?.missionDetails;
```
Comment (qdozaq) [inline src/app/dashboard/Missions/MissionEditView.jsx:183]: if you were seeing anything weird with missionDetails not showing up the DB got changed to use just `details`. 
I fixed / reflected this in the schema here https://github.com/factn/resilience-app/pull/551 so hopefully that goes in soon.

File: src/app/dashboard/Missions/MapView.jsx [hunk 5]
```diff
@@ -121,8 +121,7 @@ const Overview = ({ currentMission, missions, org, setSelectedMission, volunteer
       SpecificDetails = (
         <>
           <b>Food Box</b>
-          {mission.missionDetails?.needs?.map((box, index) => {
-            //{_.get(mission.missionDetails, "needs")?.map((box, index) => {
+          {_.get(mission.missionDetails, "needs")?.map((box, index) => {
```
Comment (jwu910) [inline src/app/dashboard/Missions/MapView.jsx]: I thought we stopped using lodash for this pattern

File: src/app/dashboard/Missions/DetailsView.jsx [hunk 6]
```diff
@@ -57,6 +57,12 @@ const useStyles = makeStyles((theme) => ({
   foodBoxDetailName: {
     padding: theme.spacing(1),
   },
+  buttonBox: {
+    spacing: theme.spacing(1),
+    display: "flex",
+    justifyContent: "center",
+    cursor: "pointer", // TODO: We shouldn't have to do this, figure out why
```
Comment (jwu910) [inline src/app/dashboard/Missions/DetailsView.jsx:64]: If you don't want to force a cursor change, the element you use needs to be an `anchor` or `button` element. something naturally clickable. But in this case you're taking an mui `Box` for this

File: src/app/dashboard/Missions/DetailsView.jsx [hunk 7]
```diff
@@ -224,13 +230,19 @@ const MissionDetailsCard = ({ mission, toEditView, toListView }) => {
...
-                Cancel Mission
-              </Box>
+              <Grid container direction="row" spacing={2} alignItems="center">
+                <Grid item>
+                  <Box item className={classes.buttonBox} onClick={toEditView}>
+                    <EditIcon className={classes.icon} /> Edit
+                  </Box>
+                </Grid>
+                <Grid item>
+                  <Box item className={classes.buttonBox} onClick={() => console.log("cancel")}>
```
Comment (jwu910) [inline src/app/dashboard/Missions/DetailsView.jsx:240]: Clicking on this box console logs?

**Comment by jwu910** [comment]
FYI @laredotornado https://github.blog/2019-02-14-introducing-draft-pull-requests/
You should be able to create draft PRs if you're looking for feedback and support working through a ticket

File: firepit-log.txt (hunk 1 above)
Comment (laredotornado) [reply firepit-log.txt]: Done!

File: src/app/dashboard/Missions/Missions.jsx (hunk 2 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/Missions.jsx]: Oh yeah, oops, reverted.

File: src/app/dashboard/Missions/MissionEditView.jsx (hunk 3 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/MissionEditView.jsx]: Ok, extraneous functions removed.

File: src/app/dashboard/Missions/MissionEditView.jsx (hunk 4 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/MissionEditView.jsx]: Thanks.  Changed to ".details"

File: src/app/dashboard/Missions/MapView.jsx (hunk 5 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/MapView.jsx]: Ah yes, this should be removed now.

File: src/app/dashboard/Missions/DetailsView.jsx (hunk 6 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/DetailsView.jsx]: Ok, switched to Buttons

File: src/app/dashboard/Missions/DetailsView.jsx (hunk 7 above)
Comment (laredotornado) [reply src/app/dashboard/Missions/DetailsView.jsx]: I'm actually not clear on what should happen when a mission is Cancelled.  Goign to ask on Slack.

**Comment by jwu910** [comment]
Here is the corresponding github user story epic.

https://github.com/factn/resilience-app/issues/431

**Pull request merged**